### ISSUE TITLE
Compatibility with gitea http implementation

### ISSUE THREAD

**Issue by reynir** [mirage/ocaml-git#601]
With #580 I had hoped I could use ocaml-git against a gitea http repository, but it fails with a 401 Unauthorized when doing a POST request at https://git.data.coop/data.coop/slides.git/git-upload-pack. Unfortunately, it loops retrying the request over and over.

I have discovered that
- it works fine with C-git
- even with smart protocol version 1 it works with C-git
- checking the curl trace from C-git I discovered they set headers `content-type`, `accept`, `git-protocol` and `user-agent`
- adding `content-type` and `accept` results in a 200 OK with an empty body

**Comment by reynir** [comment]
I spent quite some time trying to reproduce the curl invocation from C git, but I always got an empty response. I suspect it may be either binary data I copied over incorrectly (not sure that's the case), or a copy/paste mistake (the curl trace is split at every N characters). At that point I was also somewhat exhausted. I will continue trying to reproduce a working curl invocation.

I will as well submit a patch to add a content-type header as Gitea requires that: https://github.com/go-gitea/gitea/blob/dcd9fc7ee894700f702f3847d7d2a41d6a009b7e/routers/web/repo/http.go#L421
